org.spoofax.terms/src/org/spoofax/interpreter/terms/InternalIStrategoArrayList.java

```diff
@@ -0,0 +1,23 @@
+package org.spoofax.interpreter.terms;
+
+import java.io.IOException;
+import org.spoofax.terms.StrategoArrayList;
+
+/**
+ * Internal interface used to share code between the {@link StrategoArrayList} and {@link StrategoArrayListTail}
+ */
+interface InternalIStrategoArrayList {
```

Comment:
Interfaces expose internal details. If I get an `InternalIStrategoArrayList` or anything derived from it, I will be able to call the internal methods.

Reply:
Yes, that's why it all starts with `internal`, the convention is don't use anything that starts with `internal`. This isn't something I came up with btw, see `StrategoTerm#internalSetAnnotations`.

Reply:
Note that it is package private, so at least it cannot be use outside of the package.